Adding 12 Okta Identity Engine (OIE) API Functions

patrickjburke245 opened the pull request:
Had a code review with Gabriel Sroka to review PowerShell verbs, adding Okta to function definitions, and following API documentation for ID variables. For the 3 endpoints that have body parameters, I followed the definition style from Enable-OktaFactor (https://github.com/gabrielsroka/OktaAPI.psm1/blob/4667b95c3ac051b97b5683e746a3446fbd9f27fd/Modules/OktaAPI.psm1#L124-L126).

patrickjburke245 committed:
Adding 12 Okta Identity Engine (OIE) API Functions

Had a code review with Gabriel Sroka to review PowerShell verbs, adding Okta to function definitions, and following API documentation for ID variables. For the 3 endpoints that have body parameters, I followed the definition style from Enable-OktaFactor (https://github.com/gabrielsroka/OktaAPI.psm1/blob/4667b95c3ac051b97b5683e746a3446fbd9f27fd/Modules/OktaAPI.psm1#L124-L126).

gabrielsroka commented:
> For the 3 endpoints that have body parameters, I followed the definition style from Enable-OktaFactor

i see, but this is not Enable, it's New :)

try 
https://github.com/gabrielsroka/OktaAPI.psm1/blob/42070b7e06ee83359b23783bd91d7cbd12d43848/Modules/OktaAPI.psm1#L240-L242
and
https://github.com/gabrielsroka/OktaAPI.psm1/blob/42070b7e06ee83359b23783bd91d7cbd12d43848/Modules/OktaAPI.psm1#L252-L255

patrickjburke245 committed:
Update OktaAPI.psm1

Removed "body" from variable names referring to Captcha objects. This was done because this module already abstracts the HTTP request process. Also now differentiating between the Full Update and Partial Updates of "/api/v1/captchas/${id}".

patrickjburke245 commented:
Updated this PR to fix variable names and add another function. 2 more PRs coming (one for CallOktaAPI.ps1 and one for the README's Usage section).

patrickjburke245 committed:
Update OktaAPI.psm1

Added "$null = " to the 2 functions with a DELETE HTTP method to capture whatever is returned.
